textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/numbermatcher/NumberPredicate.java

```diff
@@ -0,0 +1,47 @@
+package edu.uci.ics.textdb.dataflow.numbermatcher;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IPredicate;
+import edu.uci.ics.textdb.api.storage.IDataStore;
+import edu.uci.ics.textdb.common.constants.DataConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.storage.DataReaderPredicate;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.search.MatchAllDocsQuery;
+
+/**
+ *
+ * @author Adrian Seungjin Lee
+ *
+ */
+public class NumberPredicate implements IPredicate {
+
+    private Attribute attribute;
+    private Number threshold;
+    private DataConstants.NumberMatchingType operatorType;
+
+    public NumberPredicate(Number threshold, Attribute attribute,
+                           DataConstants.NumberMatchingType operatorType) throws DataFlowException {
```

Comment:
I would like to change `operatorType` to `matchingType` to avoid confusion.
I'm aware of that `KeywordPredicate` also uses `operatorType`, I'll change it to `matchingType`  in the future.